### PR TITLE
Maxiaoyu/ordered merge join

### PR DESCRIPTION
**Pull request by ilovesoup**
Add order keeping logic so that multi-way merge join without sort is enabled.
Without a secondary sort operator, it only consider cases that required single key for now.

### CODE REVIEW

File: plan/physical_plans.go [hunk 1]
```diff
@@ -432,6 +432,43 @@ type Cache struct {
...
 
+func (p *PhysicalMergeJoin) tryConsumeOrder(prop *requiredProperty, eqCond *expression.ScalarFunction) *requiredProperty {
+	// TODO: We still can consume a partial sorted results somehow if main key matched.
+	// To do that, we need a Sort operator being able to do a secondary sort
+	if len(prop.props) != 1 || len(eqCond.GetArgs()) != 2 {
+		return prop
+	}
+
+	reqSortedColumn := prop.props[0].col
+	sortDesc := prop.props[0].desc
```
Comment (hanfei1991) [inline plan/physical_plans.go]: If sortDesc is true, return prop?

File: plan/physical_plans.go [hunk 2]
```diff
@@ -432,6 +432,43 @@ type Cache struct {
...
+		lColumn, lOk := eqCond.GetArgs()[0].(*expression.Column)
+		if lOk && lColumn.Equal(reqSortedColumn, p.ctx) && !sortDesc {
+			return removeSortOrder(prop)
+		}
+	case RightOuterJoin:
+		rColumn, rOk := eqCond.GetArgs()[1].(*expression.Column)
+		if rOk && rColumn.Equal(reqSortedColumn, p.ctx) && !sortDesc {
+			return removeSortOrder(prop)
+		}
+	default:
```
Comment (shenli) [inline plan/physical_plans.go]: The default case is usless.

File: plan/physical_plans.go (hunk 1 above)
Comment (ilovesoup) [inline plan/physical_plans.go]: Yes.That means it doesn't consume any order since we don't have desc implementation exactly.

Comment (ilovesoup) [inline plan/physical_plans.go]: done

File: plan/physical_plans.go (hunk 2 above)
Comment (ilovesoup) [inline plan/physical_plans.go]: deleted.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**